src/go/accessd/main.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	uber_zap "go.uber.org/zap"
```

Comment:
I looked into this when Mark mentioned it - glog is "frozen" for Go if I recall correctly, and this is a forked glog with fixes that the go community needed for e.g. cloud generalization (again, if I recall correctly).  Choosing this over glog made sense from what I saw.

Reply:
Also I see an implementation of our logger interface in `src/go/log/zap/zap.go` below. So why the reference here to `uber_zap`? And why our own zap implementation? Still figuring this out...

Reply:
OK, so now I think I get it. `src/go/log/zap/zap.go` is our implementation of our new `src/go/log/log.go` generalized logging interface, backed by `go.uber.org/zap`.  I'm not entirely sure how to improve the understandability here (it's not bad when you look at all the source code, but a newcomer may not do that regularly - and in a linear review of this PR the picture comes out of order). But I suppose so long as they copy pasta the logging set-up and whatnot, maybe they don't need to get the plumbing. I still think a doc on how we are doing logging in e.g. Golang Best Practices may be quite helpful.

Reply:
two features we want: 1. nested/scoped named loggers, 2. parameterized/k=v logging

the first we want because we want to be able to control log-level for various internal components. the second we want because we will one day (today?) need to elide PII in logs shared with us (e.g. IMSI)

note: we want log level changable at runtime for a particular scope, without restarting the service. glog/klog don't have the idea of a logger tree, so only provide global log level. if using glog/klog, we'd have to rebuild these functions.

lastly, zap claims to have very fast performance, 98% code coverage, and is an active project :)

Reply:
Awesome! Thanks for that. The PII thing was the biggest hurdle I faced when trying to enable sentry for our Python services. We don't scope most of our logs over there, so it made it very hard to filter to reduce load on the error level logs and scrape PII. :/